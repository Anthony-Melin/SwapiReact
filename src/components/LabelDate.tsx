import { PropsWithChildren } from 'react'

type LabelDateProps = PropsWithChildren<{
    label: string
    value: string
    testid: string
}>

const LabelDate = ({ label, value, testid, ...restProps }: LabelDateProps) => {
    const date = new Date(value)
    return (
        <p data-testid={testid} {...restProps}>
            {label}: {date.toLocaleString('fr-FR')}
        </p>
    )
}

export default LabelDate
