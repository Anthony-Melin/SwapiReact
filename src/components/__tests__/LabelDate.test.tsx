import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import LabelDate from '../LabelDate.tsx'

describe('LabelDate', () => {
    beforeEach(() => {
        render(
            <LabelDate
                label="LABEL"
                value="2014-12-09T23:50:51.644000Z"
                testid="testid"
            />
        )
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('testid')).toMatchSnapshot()
    })

    it('should render text', () => {
        expect(screen.getByTestId('testid')).toHaveTextContent(
            'LABEL: 10/12/2014 00:50:51'
        )
    })
})
