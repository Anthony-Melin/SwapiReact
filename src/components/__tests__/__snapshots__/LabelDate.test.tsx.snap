// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LabelDate > should match snapshot 1`] = `
<p
  data-testid="testid"
>
  LABEL
  : 
  10/12/2014 00:50:51
</p>
`;
